src/Simulation/DepletionVoltage.jl

```diff
@@ -0,0 +1,163 @@
+    
+    ϕmin, ϕmax = extrema((ϕρ .+ potential_range[1] * sim.weighting_potentials[contact_id].data)[inside])
+    eps = 1e-3 * sign( potential_range[length(potential_range)÷2] )
+    initial_depletion::Bool = ϕmax - ϕmin < maximum((abs(potential_range[1]), eps))
+    depletion_voltage::T = NaN
+    start_local_search::T = 0
+    
+    @showprogress for U in potential_range
+        ϕmin, ϕmax = extrema((ϕρ .+ T(U) * sim.weighting_potentials[contact_id].data)[inside])
+        depleted = ϕmax - ϕmin < abs(U)
+        if (initial_depletion && !depleted) || (!initial_depletion && depleted)
+            start_local_search = T(U)
+            break
+        end
+    end
+    local_range::AbstractRange = range(start_local_search, last(potential_range), step = step(potential_range))
+    size_data = size(sim.point_types.data)
+    
+    if size_data[2] == 1 # Fully phi symmetric detectors
+        Ix = CartesianIndex(1,0,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iz,-Iz]
+    else
+        Ix = CartesianIndex(1,0,0)
+        Iy = CartesianIndex(0,1,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iy,-Iy,Iz,-Iz]
+    end
+
+    scale = start_local_search
+    undepleted = 0
+    eps = 1e-1
+    length_idx = length(inside[1])
+    for (i,idx) in enumerate(inside)
+        for scale_loc in local_range
+            center_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[idx] + ϕρ[idx]
+            min_pot = center_pot + eps
+            max_pot = center_pot - eps
+            for neighbour in neighbours
+                n_idx = idx + neighbour
+                if 0 == sum([ (n_idx[j] == 0 || n_idx[j] > size_data[j]) for j in 1:length_idx]) # Check whether one of the indices is outside of the grid
+                    local_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[n_idx] + ϕρ[n_idx]
+                    if local_pot < min_pot
+                        min_pot = local_pot
+                    elseif local_pot > max_pot
+                        max_pot = local_pot
+                    end
+                end
+            end
+            if !(min_pot>center_pot || max_pot<center_pot) # Check whether depletion condition is met for this specific index (corresponding to center pot) and jump to next index if that is the case.
+                if abs(scale_loc)> abs(scale)
+                    scale = T(scale_loc)
+                    local_range = range(scale, last(potential_range), step = step(potential_range))
+                end         
+                break
+            end
+            if scale_loc == local_range[end] # If local scale is iterated all the way throughout the local range, this grid point is undepleted.
+                undepleted+=1
+            end         
+        end
+        if (undepleted > 0) # Break loop as soon as one grid point is undepleted
+            break
+        end
```

Comment:
Why not already break when increasing `undepleted`?

Reply:
In this case it would work, but at the moment the plan is to allow both going from depleted to undepleted and also from undepleted to depleted. Maybe this can be done there as well, but not sure yet